test/api/capi/test_capi_appender.cpp

```diff
@@ -1252,3 +1252,94 @@ TEST_CASE("Test upserting using the C API", "[capi]") {
 
 	tester.Cleanup();
 }
+
+TEST_CASE("Test clear appender data in C API", "[capi]") {
+	CAPITester tester;
+	duckdb::unique_ptr<CAPIResult> result;
+	duckdb_state status;
+
+	REQUIRE(tester.OpenDatabase(nullptr));
+
+	// create a table and insert initial data
+	REQUIRE_NO_FAIL(tester.Query("CREATE TABLE integers(i INTEGER)"));
+	REQUIRE_NO_FAIL(tester.Query("INSERT INTO integers VALUES (1)"));
+
+	// create appender and append many rows
+	duckdb_appender appender;
+	status = duckdb_appender_create(tester.connection, nullptr, "integers", &appender);
+	REQUIRE(status == DuckDBSuccess);
+	REQUIRE(duckdb_appender_error(appender) == nullptr);
+
+	// append a bunch of values that should be cleared
+	for (idx_t i = 0; i < 2000; i++) {
```

Comment:
Let's actually append a bit over 2048, to ensure we're flushing a full chunk to the column data collection.